### PR TITLE
chore: read trace parent

### LINKED ISSUE #19: context deadline exceeded for big XML files
The attached file needs to be renamed to XML, as GH does not allow attaching XML files

[TEST-NotificationManagerStepTests.txt](https://github.com/mdelapenya/junit2otlp/files/8060723/TEST-NotificationManagerStepTests.txt)

### PR DESCRIPTION
**Pull request by mdelapenya**
- chore: extract traceparent from environment
- docs: document TRACEPARENT
- fix: increase timeout when cancelling metrics pusher (Closes #19)

## What does this PR do?
It initializes the context to be aware of possible parent traces, using the TRACEPARENT environment variable.

It also fixes a cancelling timeout when initializing the metrics pusher

## Why is it important?
The tool should create the distributed traces for the tests, using its parent if available

### CODE REVIEW

File: traceenv.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	// open-telemetry/opentelemetry-specification#740
+	parent := os.Getenv("TRACEPARENT")
+	state := os.Getenv("TRACESTATE")
+
+	if parent != "" {
+		tc := propagation.TraceContext{}
+		return tc.Extract(ctx, &textMap{parent: parent, state: state})
+	}
+
+	// deprecated: removed in v0.11.0
```
Comment (kuisathaverat) [inline traceenv.go:26]: Do worth adding deprecated stuff?

**Review by kuisathaverat — APPROVED** [review]

File: traceenv.go (hunk 1 above)
Comment (mdelapenya) [reply traceenv.go:26]: Extracted this entire block from Moby's buildkit. Maybe we can simply exclude that part, as we have a strong dependency on the opentelemetry SDK v1.1.0

Comment (mdelapenya) [reply traceenv.go]: Removed, thanks!

File: traceenv.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	tracestateHeader  = "tracestate"
+)
+
+// see https://github.com/moby/buildkit/pull/2572
+func initOtelContext(ctx context.Context) context.Context {
+	// open-telemetry/opentelemetry-specification#740
+	parent := os.Getenv("TRACEPARENT")
+	state := os.Getenv("TRACESTATE")
+
+	if parent != "" {
```
Comment (v1v) [inline traceenv.go:21]: what if `state` is empty?

Comment (mdelapenya) [reply traceenv.go:21]: It's ignored: https://github.com/open-telemetry/opentelemetry-go/blob/v1.1.0/propagation/trace_context.go#L142-L145

**Comment by mdelapenya** [comment]
@v1v I'm merging this one and test in our shared library its usage. I already addressed your comments about the state being empty

**Pull request merged**